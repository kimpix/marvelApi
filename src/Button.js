import React, { Component } from 'react';

const Button = (props) => {
    return (
      <button onClick={props.onClick} className={props.class}>{props.button}</button>
    )
};

export default Button;