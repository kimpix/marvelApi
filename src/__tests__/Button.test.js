import React from 'react';
import Button from '../Button';


test('Button render', (props ={}) => {

    const tree = shallow(
      <Button onClick={jest.fn()} class={jest.fn()} button={jest.fn()}/>
    );

    expect(tree).toMatchSnapshot();
});


test('render button class', (props ={}) => {
    const tree = shallow(
      <Button onClick={jest.fn()} class='infinite' button={jest.fn()}/>
    );

    const dom = tree.props().className;

    expect(dom).toEqual('infinite');
});


test('render button text', (props ={}) => {
    const tree = shallow(
      <Button onClick={jest.fn()} class={jest.fn()} button='infinite is the way'/>
    );

    const dom = tree.props().children;

    expect(dom).toEqual('infinite is the way');
});

