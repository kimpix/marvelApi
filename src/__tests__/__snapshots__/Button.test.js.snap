// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<button
  className={[MockFunction]}
  onClick={[MockFunction]}
/>
`;
